sklearn/utils/testing.py

```diff
@@ -882,3 +903,156 @@ def check_docstring_parameters(func, doc=None, ignore=None, class_name=None):
             if n1 != n2:
                 incorrect += [func_name + ' ' + n1 + ' != ' + n2]
     return incorrect
+
+
+def _check_matching_docstrings(doc_list, type_dict, type_name, object_name,
+                               include, exclude):
+    """
+    Checks if the docstring element, Parameter/Attribute/Return, having the
+    same name as a key in ``type_dict``, also has the same type definition and
+    description as that in ``type_dict``.
+
+    If a matching key is not found in ``type_dict``, the docstring element is
+    added in it with it's name as the key and value being a dictionary of it's
+    type_definition and description.
+
+    """
+    for name, type_definition, description in doc_list:
+        if exclude is not None and name in exclude:
+            pass
+        elif include is not True and name not in include:
+            pass
+        else:
+            # remove all whitespaces
+            type_definition = " ".join(type_definition.split())
+            description = [" ".join(s.split()) for s in description]
+            description = list(filter(None, description))
+
+            if name in type_dict:
+                u_dict = type_dict[name]
+                msg1 = (type_name + " '" + name + "' of '" + object_name +
+                        "' has inconsistent type definition with that of '" +
+                        u_dict['object'] + "'.")
+                msg2 = (type_name + " '" + name + "' of '" + object_name +
+                        "' has inconsistent description with that of '" +
+                        u_dict['object'] + "'.")
+
+                assert u_dict['type_definition'] == type_definition, msg1
+                assert u_dict['description'] == description, msg2
+            else:
+                add_dict = {'type_definition': type_definition,
+                            'description': description,
+                            'object': object_name}
+                type_dict[name] = add_dict
+
+    return type_dict
+
+
+def assert_consistent_docs(objects,
+                           include_params=True, exclude_params=None,
+                           include_attribs=True, exclude_attribs=None,
+                           include_returns=True, exclude_returns=None):
+    """
+    Checks consistency between the docstring of ``objects``.
+
+    Checks if types and descriptions of Parameters/Attributes/Returns are
+    identical across ``objects``. Raises AssertionError if found otherwise.
+
+    Parameters
+    ----------
+    objects : list
+        The list of objects that may be either ``NumpyDocString`` instances or
+        objects (classes, functions, descriptors) with docstrings that can be
+        parsed as numpydoc.
+
+    include_params : list, False or True (default)
```

Comment:
could change list -> collection. All we care is that `in` works (or in some other implementation, iteration may be used; but still, a collection is sufficient).

Reply:
So I should just allow collections to be passed as arguments. Wont it break somewhere? It might need some testing as well. What do you say?